stream_alert/athena_partition_refresh/main.py

```diff
@@ -0,0 +1,206 @@
+
+    Returns:
+        [string]: The result of the query, this can be SUCCEEDED, FAILED, or CANCELLED.
+                  Reference http://bit.ly/2uuRtda
+    """
+    @backoff.on_predicate(backoff.fibo,
+                          lambda status: status in ('QUEUED', 'RUNNING'),
+                          max_value=10,
+                          jitter=backoff.full_jitter,
+                          on_backoff=_backoff_handler,
+                          on_success=_success_handler)
+    def _get_query_execution(query_execution_id):
+        return ATHENA_CLIENT.get_query_execution(
+            QueryExecutionId=query_execution_id
+        )['QueryExecution']['Status']['State']
+
+    return _get_query_execution(query_execution_id)
+
+
+def run_athena_query(**kwargs):
+    """Helper function to run Athena queries
+
+    Keyword Args:
+        query [string]: The SQL query to execute
+        database [string]: The database to execute the query against
+        results_bucket [string]: The S3 bucket to store query results
+        results_path [string]: The S3 key to store results
+
+    Returns (one or the other):
+        [bool]: Query success
+        [dict]: Query result response
+    """
+    LOGGER.debug('Executing query: %s', kwargs['query'])
+    query_execution_resp = ATHENA_CLIENT.start_query_execution(
+        QueryString=kwargs['query'],
+        QueryExecutionContext={'Database': kwargs.get('database', DATABASE_DEFAULT)},
+        ResultConfiguration={'OutputLocation': '{}/{}'.format(
+            kwargs['results_bucket'],
+            kwargs['results_path']
+        )}
+    )
+    query_execution_result = check_query_status(query_execution_resp['QueryExecutionId'])
+    if query_execution_result != 'SUCCEEDED':
+        LOGGER.error(
+            'The query %s returned %s, exiting!',
+            kwargs['query'],
+            query_execution_result)
+        return False
+
+    query_results_resp = ATHENA_CLIENT.get_query_results(
+        QueryExecutionId=query_execution_resp['QueryExecutionId'],
+    )
+
+    # The idea here is to leave the processing logic to the calling functions.
+    # No data being returned isn't always an indication that something is wrong.
+    if not query_results_resp['ResultSet']['Rows']:
+        LOGGER.debug('The query %s returned no data', kwargs['query'])
+
+    return query_results_resp
+
+
+def check_database_exists(results_bucket, results_path):
+    """Verify the StreamAlert Athena database exists."""
+    query_resp = run_athena_query(
+        query='SHOW DATABASES LIKE \'streamalert\';',
+        results_bucket=results_bucket,
+        results_path=results_path
+    )
+    if isinstance(query_resp, dict) and not query_resp['ResultSet']['Rows']:
+        LOGGER.info('The \'streamalert\' database does not exist, please create it.')
+        return False
+
+    return True
+
+
+def check_table_exists(results_bucket, results_path, table_name):
```

Comment:
It looks like this function is only used in the CLI (not in the Lambda function). If that's the case, consider factoring it out into the CLI code (unless you plan on using this later)